compiler/rustc_const_eval/src/interpret/operand.rs

```diff
@@ -631,6 +631,9 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         let mut op = self.local_to_op(self.frame(), mir_place.local, layout)?;
         // Using `try_fold` turned out to be bad for performance, hence the loop.
         for elem in mir_place.projection.iter() {
+            if elem.is_subtype() {
+                continue;
+            }
```

Comment:
This seems suspicious. The type at the end might not match what it says in the MIR. Why is this needed?

Reply:
Otherwise we get `No `move_errors` should be allowed in MIR borrowck`, which shouldn't happen(?) with changes introduced in this PR.

Reply:
This is in the interpreter. There's no way this can cause borrowck-related errors. Also you wrote that same comment in reply to two different threads here so I assume this is a copy-paste error.

Reply:
Well that's error I got but I will investigate it further as it seems there is something wrong going on here. Yeah sorry I replied twice to two different threads.

Reply:
Yeah that sounds very wrong. If this is a subtype projection it should just go to `project_subtype` and then it should do its thing from there.